File: DotNet/Messaging/MessageIds/MessageIdManager.cs
using System;

namespace Bollywell.Hydra.Messaging.MessageIds
{
    /// <summary>
    /// Methods for manipulating IMessageIds.
    /// </summary>
    public static partial class MessageIdManager
    {
        public static IMessageId Create(string couchId)
        {
            return new LongMessageId(couchId);
        }

        public static IMessageId Create(DateTime utcDate)
        {
            return new LongMessageId(utcDate);
        }

        public static bool IsMessageId(string couchId)
        {
            return LongMessageId.IsMessageId(couchId);
        }

    }
}

File: Java/hydra/src/main/java/uk/co/shastra/hydra/messaging/messageids/MessageIdManager.java
package uk.co.shastra.hydra.messaging.messageids;

import java.util.Date;

public class MessageIdManager {

    public static MessageId create(String couchId)
    {
        return new UtcIdMessageId(couchId);
    }

    public static MessageId create(Date utcDate)
    {
        return new UtcIdMessageId(utcDate);
    }

    public static boolean isMessageId(String couchId)
    {
        return UtcIdMessageId.isMessageId(couchId);
    }
    
}
